### ISSUE TITLE
Allows to split view 3D and 2D together

### ISSUE THREAD

**Issue by Hanprogramer** [godotengine/godot-proposals#6713]
### Describe the project you are working on

A 3D RPG Top-down View. I'm using this feature to develop the inventory.

### Describe the problem or limitation you are having in your project

Currently in the editor the splitting part is already implemented, but it can't be one for 2D and one for 3D. This is especially useful for me because I have to position my camera in a SubViewport (GD4) but I can't see it directly.
![image](https://user-images.githubusercontent.com/30620423/232551425-91974441-7cf3-4111-bbbb-701f02dc0e86.png)
![image](https://user-images.githubusercontent.com/30620423/232551568-aa4e3e9d-00a6-4fb1-8e7a-8fe97b7eaf4c.png)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The feature is to allow one side of the panel to display 2D and the other to display 3D. I know this might require a lot of work to implement.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

My idea is this:
![image](https://user-images.githubusercontent.com/30620423/232551901-ed871e1a-f87d-416a-be23-a24c6b624d4c.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't think so. Since this requires editing almost a very core part of the editor. 

### Is there a reason why this should be core and not an add-on in the asset library?

This is related to the core of the editor and would be very hard to implement via an asset library/addons

**Comment by yosoyfreeman** [comment]
Now that we are talking about this, being able to have 3D + Script editor would be super nice for working with tool scripts or procedurally generated content.

**Comment by Hanprogramer** [comment]
> Now that we are talking about this, being able to have 3D + Script editor would be super nice for working with tool scripts or procedurally generated content.

Or even better, instead of having the top buttons (3D, 2D, Scripts, Assets) why not just tabs? So every tab will be based on what it's opened. For example the scenes will open a scene editor which have a toggle for 2D and 3D. A script tab will be used to edit scripts per tab. Assets manager will be a separate tabs as well. This opens up so many possibilities (for example docking) but requires a big change to the editor and how it works but i think it is very worth to do.

**Comment by Hanprogramer** [comment]
Oh and on top of this issue, I think it's also better to add a button to display the viewports, like how the camera has "preview" button, maybe somehow viewports can have them especially if they own 3D world on their own.

**Comment by yosoyfreeman** [comment]
Yeah, i'm so used to other engines that i forgot i can just play the game in a standalone window and the code and everything updates. I think godot has this fixed by design

**Comment by Amealky** [comment]
Dunno if it can help you but I made this little plugin recently its my first one and I also made a proposal about that but it could be made by plugin way I think

here is the link : https://github.com/Amealky/Previewer_2D_3D

**Comment by Meta-Ben** [comment]
Don't know if it's okay to update this issue but the plugin I made on my previous post evolve with the help of a contributor ( big thanks to him ! ) 

And its now possible to work on 2D and 3D in the splitted view ( not only preview the other ) @Hanprogramer 

new link : https://github.com/Meta-Ben/DualEditor
